react/src/components/tabs/components/SprkTabsButton/SprkTabsButton.test.js

```diff
@@ -25,41 +33,128 @@ describe('SprkTabsButton:', () => {
+        tabBtnChildren="Tab 1"
+        isFocused
+        tabBtnId="tab-52"
+        tabBtnAdditionalClasses="test"
+      >
+        Test Content 1
+      </SprkTabsButton>,
+    );
     wrapper.find('.sprk-c-Tabs__button').simulate('click');
     expect(wrapper.find('.sprk-c-Tabs__button.test').length).toBe(1);
   });
 
-  it('should focus the next Tab Button when the right arrow key is pressed', () => {
+  it(`should focus the next Tab Button when the right arrow key
+  is pressed`, () => {
     const wrapper = mount(
       <SprkTabs>
-        <SprkTabsPanel isDefaultActive tabBtnChildren="Tab 1">Test Content 1</SprkTabsPanel>
+        <SprkTabsPanel isDefaultActive tabBtnChildren="Tab 1">
+          Test Content 1
+        </SprkTabsPanel>
         <SprkTabsPanel tabBtnChildren="Tab 2">Test Content 2</SprkTabsPanel>
       </SprkTabs>,
     );
     // First, tab into the Tab Button area
-    wrapper.find('div.sprk-c-Tabs__buttons').simulate('keydown', { keyCode: 9 });
+    wrapper
+      .find('div.sprk-c-Tabs__buttons')
+      .simulate('keydown', { keyCode: 9 });
     // Second, hit the right arrow key
-    wrapper.find('div.sprk-c-Tabs__buttons').simulate('keydown', { keyCode: 39 });
+    wrapper
+      .find('div.sprk-c-Tabs__buttons')
+      .simulate('keydown', { keyCode: 39 });
     const button = wrapper.find(SprkTabsButton).last();
-    expect(button.getDOMNode().classList.contains('sprk-c-Tabs__button--active')).toBe(true);
+    expect(
+      button.getDOMNode().classList.contains('sprk-c-Tabs__button--active'),
+    ).toBe(true);
+  });
+
+  // TODO remove as part of Issue 3566
+  it('should add additional classes from deprecated tabBtnAddClasses', () => {
+    const wrapper = mount(
+      <SprkTabsButton tabBtnAddClasses="test">Test Content 1</SprkTabsButton>,
+    );
+    expect(wrapper.find('.sprk-c-Tabs__button.test').length).toBe(1);
+  });
+
+  it('should add additional classes from tabBtnAdditionalClasses', () => {
+    const wrapper = mount(
+      <SprkTabsButton tabBtnAdditionalClasses="test">
+        Test Content 1
+      </SprkTabsButton>,
+    );
+    expect(wrapper.find('.sprk-c-Tabs__button.test').length).toBe(1);
+  });
+
+  // TODO remove as part of Issue 3566
+  it('should prefer tabBtnAdditionalClasses over tabBtnAddClasses', () => {
+    const wrapper = mount(
+      <SprkTabsButton tabBtnAdditionalClasses="test" tabBtnAddClasses="test2">
+        Test Content 1
+      </SprkTabsButton>,
+    );
+    expect(wrapper.find('.sprk-c-Tabs__button.test').length).toBe(1);
+    expect(wrapper.find('.sprk-c-Tabs__button.test2').length).toBe(0);
+  });
+
+  it('should add data-analytics from tabBtnAnalytics', () => {
+    const wrapper = mount(
+      <SprkTabsButton tabBtnAnalytics="test">Test Content 1</SprkTabsButton>,
+    );
+    expect(
+      wrapper.find('.sprk-c-Tabs__button[data-analytics="test"]').length,
+    ).toBe(1);
+  });
+
+  it('should prefer tabBtnAnalyticsString over tabBtnAnalytics', () => {
```

Comment:
Let's add a stand alone test for `tabBtnAnalyticsString` and add a TODO to remove the `tabBtnAnalytics` test